specification/postgresqlhsc/resource-manager/Microsoft.DBforPostgreSQL/preview/2023-03-02-preview/postgresqlhsc.json

```diff
@@ -2322,6 +2334,31 @@
         }
       }
     },
+    "DataEncryption": {
+      "type": "object",
+      "description": "The data encryption properties of a cluster.",
+      "properties": {
+        "primaryKeyUri": {
+          "type": "string",
+          "description": "URI for the key in keyvault for data encryption of the primary server."
+        },
+        "primaryUserAssignedIdentityId": {
+          "type": "string",
+          "description": "Resource Id for the User assigned identity to be used for data encryption of the primary server."
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "AzureKeyVault",
+            "SystemAssigned"
+          ],
+          "x-ms-enum": {
+            "name": "DataEncryptionType",
+            "modelAsString": true
+          }
+        }
+      }
+    },
```

Comment:
I am curious about the difference between this contract and the one here:

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/customermanagedkeys.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/customermanagedkeys.json)

it seems like are sort of for the same general purpose, i.e. using a customer supplied key to do data encryption.

Is there any significant difference in the scenario?